src/app/client/src/app/modules/public/module/player/components/public-collection-player/public-collection-player.component.html

```diff
@@ -2,26 +2,37 @@
   <div class="ui stackable grid">
 
     <!--Content Section-->
-    <div class="twelve wide column" [appTelemetryImpression]="telemetryContentImpression" id="app-player-collection-renderer" *ngIf="showPlayer">
+    <div class="twelve wide column" [appTelemetryImpression]="telemetryContentImpression" id="app-player-collection-renderer" *ngIf="showPlayer && playerContent">
 
       <!--Content Title-->
       <div class="sb-pageSection">
         <div class="sb-pageSection-header">
           <h1 class="sb-pageSection-title m-0 mr-8">
-            {{contentTitle}}
+            {{playerContent.name}}
```

Comment:
we can add this in content-download component and use the same component in all 3 places. This component will handle download and update